src/include/assert.h

```diff
@@ -62,17 +62,25 @@ struct BackTrace;
 # define __CEPH_ASSERT_FUNCTION ((__const char *) 0)
 #endif
 
+#ifdef __cplusplus
 extern void register_assert_context(CephContext *cct);
+#endif
+
 extern void __ceph_assert_fail(const char *assertion, const char *file, int line, const char *function)
   __attribute__ ((__noreturn__));
 extern void __ceph_assertf_fail(const char *assertion, const char *file, int line, const char *function, const char* msg, ...)
   __attribute__ ((__noreturn__));
 extern void __ceph_assert_warn(const char *assertion, const char *file, int line, const char *function);
 
+#ifdef __cplusplus
+# define __CEPH_ASSERT_VOID_CAST static_cast<void>
+#else
+# define __CEPH_ASSERT_VOID_CAST (void)
```

Comment:
nit: identifiers starting with `__` are [reserved](http://en.cppreference.com/w/cpp/language/identifiers) and their use is _technically_ undefined behavior